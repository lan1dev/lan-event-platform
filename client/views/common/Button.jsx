import styled from 'styled-components';
import colors from '../constants/colors';
import fonts from '../constants/fonts';

/* eslint-disable react/destructuring-assignment */
const Button = styled.button`
  background: ${colors.primary};
  border-color: ${colors.dark};
  color: ${colors.dark};
  line-height: 1.5;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid;
  white-space: nowrap;
  padding: 0 15px;
  font-size: 14px;
  border-radius: 4px;
  height: 32px;
  user-select: none;
  transition: all 0.3s cubic-bezier(0.645, 0.045, 0.355, 1);
  position: relative;
  font-family: ${fonts.primary};

  &:hover {
    background: ${colors.dark};
    color: ${colors.white};
    border-color: ${colors.dark};
  }
`;
/* eslint-enable react/destructuring-assignment */

export default Button;
