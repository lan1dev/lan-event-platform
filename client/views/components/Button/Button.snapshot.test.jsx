import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import Button from './index';

describe('Button', () => {
  let props;
  let wrapper;

  const Wrapper = () => {
    if (!wrapper) {
      wrapper = shallow(<Button {...props}>{props.children}</Button>);
    }
    return wrapper;
  };

  // Reset Component and Props before each test.
  beforeEach(() => {
    props = {};
    wrapper = undefined;
  });

  describe('without props', () => {
    it('Should match snapshot', () => {
      expect(toJson(Wrapper())).toMatchSnapshot();
    });
  });
});
