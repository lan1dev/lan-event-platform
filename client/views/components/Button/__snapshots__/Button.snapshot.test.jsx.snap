// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button without props Should match snapshot 1`] = `
.c0 {
  background: white;
  border-color: gray;
  color: gray;
  line-height: 1.5;
  display: inline-block;
  font-weight: 400;
  text-align: center;
  touch-action: manipulation;
  cursor: pointer;
  background-image: none;
  border: 1px solid;
  white-space: nowrap;
  padding: 0 15px;
  font-size: 14px;
  border-radius: 4px;
  height: 32px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  transition: all 0.3s cubic-bezier(0.645,0.045,0.355,1);
  position: relative;
  font-family: Poppins;
}

.c0:hover {
  background: gray;
  color: #fff;
  border-color: gray;
}

<button
  className="c0"
/>
`;
